### ISSUE TITLE
[Bug] Swapping to a fighter after launching and recalling it from a carrier results in a broken game.

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#6519]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When switching to flying a fighter that I have previously launched and recovered as a flagship, the game runs into a problem trying to launch the ship, and ends up launching, but without the ship. (see picture below).

Note: I can buy a new fighter and launch as that just fine, but not when I am flying a fighter that had previously been carried.

On what might be a related note, any fighter that I fly as my flagship seems to lose the ability to be carried after that. I can't switch back to a carrier as my flagship and have any fighter that I have flown stored in the bays. They'll take off as escorts, but refuse to dock. (and don't show up indented anymore either)

### Steps to Reproduce

1. Bought a carrier and a finch. Carrier is my flagship. I take off fine, can launch and recover finch. All is good.
2. Landed, moved finch into flagship position, and parked the carrier. 
3. I take off, and there's no warning about hyperspace capabilities, but my ship never appears. I am sitting here staring at the omnis ringworld, with my HUD saying I'm flying a finch, but nothing is visible, I can't move, and I can't even land again. 

4. I escape to the title screen, reload my character (the current one, this is *not* going back to a previous snapshot or save).
5. Now I can launch in the fighter and fly around just fine.
6. I land and switch back to the carrier as flagship.
7. I take off, and it gives me the warning about having a ship in my fleet without hyperspace capabilities. 
8. Once I'm in space, I note that my finch is following me around, already launched like an escort. 
9. I try to recall it, and can't. The finch refuses to come onboard. It obeys commands to gather, hold position, and target. But it will not enter a bay on my carrier. 
10. I swap to flying the finch as my flagship with the carrier following, and try to board the carrier, and this also doesn't work.
11. I look at my player info screen, and the finch is no longer indented the way fighters usually are.

I have repeated this exact process with two separate Ibis & petrel pairs as well as with a carrier & finch pair.

### Expected Behavior

1. That when I switch to a fighter, I should be able to take off in it succesfully, and
2. That once I land, said fighter should be able to be carried in any ship with appropriate bays.

### Screenshots

1 - Here's my pair of ships before launching. Everything looks good.
![1 - Pre launch](https://user-images.githubusercontent.com/32169904/151706854-d38cb3bc-4e19-416f-8bde-6f6139ad6afe.PNG)

2 - Here is me  ditching the carrier to launch as the fighter. (this is after having launched as the carrier, deployed and retrieved the fighter, and landed again)
![2 - Launching as a fighter](https://user-images.githubusercontent.com/32169904/151706875-98d9f1c9-180a-4c08-bbe6-f3ae77011c4f.PNG)

3 - Here is the result of me trying to take off with the fighter as flagship. (I actually had the carrier and petrel active too. They just moved quickly off the top of the screen)
![3- Launching as the Finch (didn't reach orbit)](https://user-images.githubusercontent.com/32169904/151706938-3e85eb2b-22cf-4925-8ca7-e730415c378a.PNG)

4- After reloading the save, I can successfully launch in the fighter - but it isn't indented anymore.
![4 - No more indentation for the fighter](https://user-images.githubusercontent.com/32169904/151706966-682d5834-a318-4ffd-bd22-d2d1adc22672.PNG)

### Link to save file

[Janice Testerv03.txt](https://github.com/endless-sky/endless-sky/files/7966288/Janice.Testerv03.txt)

Note: This is currently using Omnis for testing purposes, and is currently in the main Omnis system.

### Operating System

Windows 10 Pro

### Game Source

Other (please describe)

### Game Version

70ed557 (pulled from one of my content PR's artifacts, updated to latest commit as of 2022-Jan-30)

### Additional Information
There is some conversation in the discord bug channel about this and some similar sort of stuff:
https://discord.com/channels/251118043411775489/536900466655887360/937366653841334332

### Possibly Related
https://github.com/endless-sky/endless-sky/pull/6323 Given that this deals specifically with the fighter-to-carrier transition, might it include something useful to help resolve this problem?

**Comment by vitalchip** [comment]
Two problems at work here.

Addressing point 3 - Deploying and recalling the Finch sets "shouldRemove" but this isn't unset when unloading bays, only when launching. So when you make it the flagship it's still "removed" and so doesn't even zoom off the planet and since zoom < 1 it can't act. The same thing happens if you park the carrier and depart in a ship with no bays, the fighter should just launch along with you but it's not there.

Calling UnmarkForRemoval() in Unload Bays() fixes this but I haven't tested fully for any unwanted behaviours, might be an issue for Engine::Place (npc)

<img width="419" alt="Screenshot 2022-01-30 at 19 30 22" src="https://user-images.githubusercontent.com/65342577/151714503-44c3f9d5-0d84-401c-a40f-14cfa624c884.png">


Addressing point 7 - To use a fighter as flagship AllowCarried is set false but as far aas I can see nothing sets it true when the flagship is swapped, so when you launch the Finch comes along but it's now a ship which cannot be carried, so you get the hyperspace warning and the not docking. This is also the explaination for point 11 as the indenting is done by calling CanBeCarried.

Addressing point 10 - I don't recall this being current functionality. We'd need #6323

**Comment by Zitchas** [comment]
Thanks for digging into that! I appreciate the explanations.

**Comment by petervdmeer** [comment]
Added 0.9.15 milestone, departing and not seeing the ship appearing is quite a serious bug.
